From 769c2daaaf02b0fb10da83499c76d6563b5f2ba8 Mon Sep 17 00:00:00 2001
From: Sven van Haastregt <sven.vanhaastregt@arm.com>
Date: Mon, 20 Jun 2022 11:25:54 +0100
Subject: [PATCH 054/172] Fix build by reverting a ValueMap back to std::map

Commit 4a9c78ee ("Prepare SPIRVWriter for type conversion without
opaque pointers. (#1499)", 2022-06-20) changed
`OCLTypeToSPIRVBase::AdaptedTy` to a `ValueMap`, but that made
`OCLTypeToSPIRVBase` non-movable and non-copyable.  Landing eb8e9adc
("Migrate SPIRVWriter to new PassManager", 2022-06-20) on top caused a
build breakage.

(cherry picked from commit 2a8f52469527346c8cc3cf12371a2fe7592e285f)
---
 lib/SPIRV/OCLTypeToSPIRV.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/SPIRV/OCLTypeToSPIRV.h b/lib/SPIRV/OCLTypeToSPIRV.h
index 1d135c86..17d3e7dc 100644
--- a/lib/SPIRV/OCLTypeToSPIRV.h
+++ b/lib/SPIRV/OCLTypeToSPIRV.h
@@ -46,7 +46,6 @@
 #include "llvm/IR/Function.h"
 #include "llvm/IR/LLVMContext.h"
 #include "llvm/IR/PassManager.h"
-#include "llvm/IR/ValueMap.h"
 #include "llvm/Pass.h"
 
 #include <map>
@@ -71,7 +70,7 @@ private:
   llvm::Module *M;
   llvm::LLVMContext *Ctx;
   // Map of argument/Function -> {pointee type, address space}
-  llvm::ValueMap<llvm::Value *, std::pair<llvm::Type *, unsigned>> AdaptedTy;
+  std::map<llvm::Value *, std::pair<llvm::Type *, unsigned>> AdaptedTy;
   std::set<llvm::Function *> WorkSet; // Functions to be adapted
 
   void adaptFunctionArguments(llvm::Function *F);
-- 
2.20.1

